src/lib/data/attachment.ts

```diff
@@ -0,0 +1,344 @@
+import { isRight } from 'fp-ts/lib/Either';
+import * as t from 'io-ts';
+import { AttachmentId } from '../../db/models/attachment';
+import {
+  AttachmentPrototypeId,
+  AttachmentType,
+} from '../../db/models/attachmentPrototype';
+import { GoverningEntityId } from '../../db/models/governingEntity';
+import { ATTACHMENT_VERSION_VALUE } from '../../db/models/json/attachment';
+import { PlanId } from '../../db/models/plan';
+import { PlanEntityId } from '../../db/models/planEntity';
+import { Database } from '../../db/type';
+import { InstanceDataOfModel } from '../../db/util/raw-model';
+import { organizeObjectsByUniqueProperty } from '../../util';
+import { ioTsErrorFormatter } from '../../util/io-ts';
+import { createBrandedValue } from '../../util/types';
+import { SharedLogContext } from '../logging';
+import { MapOfGoverningEntities } from './governingEntities';
+import {
+  calculateReflectiveTransitiveEntitySupport,
+  ValidatedPlanEntities,
+} from './planEntities';
+
+const ATTACHMENT_DATA = t.union([
+  t.type({
+    type: t.literal('caseLoad'),
+    value: ATTACHMENT_VERSION_VALUE['caseLoad'],
+  }),
+  t.type({
+    type: t.literal('contact'),
+    value: ATTACHMENT_VERSION_VALUE['contact'],
+  }),
+  t.type({
+    type: t.literal('cost'),
+    value: ATTACHMENT_VERSION_VALUE['cost'],
+  }),
+  t.type({
+    type: t.literal('fileWebContent'),
+    value: ATTACHMENT_VERSION_VALUE['fileWebContent'],
+  }),
+  t.type({
+    type: t.literal('indicator'),
+    value: ATTACHMENT_VERSION_VALUE['indicator'],
+  }),
+  t.type({
+    type: t.literal('textWebContent'),
+    value: ATTACHMENT_VERSION_VALUE['textWebContent'],
+  }),
+]);
+
+type PlanAttachmentData = t.TypeOf<typeof ATTACHMENT_DATA>;
+
+type PlanAttachmentParent =
```

Comment:
Again, reference to plan attachment is misleading. All attachments are attached to plan, but not all attachments are plan attachments :smile: This type illustrates it well, but its name is not reflecting that
```suggestion
type AttachmentParent =
```